src/App.test.js

```diff
@@ -63,26 +82,33 @@ describe('App', () => {
   });
 
   it('should allow users to click on a movie and take them to a Movie Preview', async () => {
-    getSingleMovieData.mockResolvedValueOnce({
-      movie: {
-        id: 337401,
-        title: 'Mulan',
-        poster_path:
-          'https://image.tmdb.org/t/p/original//aKx1ARwG55zZ0GpRvU2WrGrCG9o.jpg',
-        backdrop_path:
-          'https://image.tmdb.org/t/p/original//zzWGRw277MNoCs3zhyG3YmYQsXv.jpg',
-        release_date: '2020-09-04',
-        overview:
-          'When the Emperor of China issues a decree that one man per family must serve in the Imperial Chinese Army to defend the country from Huns, Hua Mulan, the eldest daughter of an honored warrior, steps in to take the place of her ailing father. She is spirited, determined and quick on her feet. Disguised as a man by the name of Hua Jun, she is tested every step of the way and must harness her innermost strength and embrace her true potential.',
-        genres: ['Action', 'Adventure', 'Drama', 'Fantasy'],
-        budget: 200000000,
-        revenue: 57000000,
-        runtime: 115,
-        tagline: '',
-        average_rating: 4.909090909090909,
-      },
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+
+    const movieCard = await waitFor(() =>
+      screen.getByTestId('moviecard-element-337401')
+    );
+    fireEvent.click(movieCard);
+
+    await waitFor(() => {
+      expect(screen.getByTestId('movie-preview-page')).toBeInTheDocument();
     });
+  });
+  it('should show a App component for / router', async () => {
```

Comment:
I learned something new today with these tests for sure! Using Route and Paths inside the MemoryRouter were new to me